tests/test_fact_manager.py

```diff
@@ -0,0 +1,394 @@
+                                      f" CONSTRAINT pytest_log_pkey PRIMARY KEY (id))")
+
+    test_data = [f"INSERT INTO {test_table} (name, lang, message, author, editedby, edited) "
+                 f"VALUES ('stats', 'en', 'Fuel Rats Statistics: https://t.fuelr.at/stats',"
+                 f" 'CrunchyBaton955', 'CrunchyBaton955)', '{datetime.datetime(1970, 1, 1, 0, 0, 0)}')",
+                 f"INSERT INTO {test_table} (name, lang, message, author, editedby, edited) "
+                 f"VALUES ('test', 'en', 'This is a test fact.', 'Shatt', 'Shatt',"
+                 f" '{datetime.datetime(1970, 1, 1, 0, 0, 0)}')",
+                 f"INSERT INTO {test_table} (name, lang, message, author, editedby, edited) "
+                 f"VALUES ('test', 'ru', 'This is a test fact. Чтобы зажечь "
+                 f"маяк, зажмите клавижу Х и нажмите', 'Shatt', 'Shatt',"
+                 f" '{datetime.datetime(1970, 1, 1, 0, 0, 0)}')"]
+
+    pytest_fm = FactManager(fact_table=test_table, fact_log=test_log)
+
+    # Let's add our test data:
+    with pytest_fm._dbpool.getconn() as conn:
+        conn.autocommit = True
+        conn.set_client_encoding('utf-8')
+
+        with conn.cursor() as cursor:
+            cursor.execute(test_table_creation)
+            cursor.execute(test_log_table_creation)
+            # Add test data:
+            for data in test_data:
+                cursor.execute(data)
+
+    def teardown():
+        # Tear everything down
+        with pytest_fm._dbpool.getconn() as conn:
+            conn.autocommit = True
+            conn.set_client_encoding('utf-8')
+
+            with conn.cursor() as cursor:
+                cursor.execute(f"DROP TABLE {test_table}")
+                cursor.execute(f"DROP TABLE {test_log}")
+
+    request.addfinalizer(teardown)
+    return pytest_fm
+
+
+def test_module_typeerror_fact_table(test_fm_fx):
+    """
+    Verify a TypeError is thrown, if assigning a non-str value to an optional property.
+    """
+    with pytest.raises(TypeError):
+        testFM = FactManager(fact_table=["a list"])
+
+
+def test_module_typeerror_log_table(test_fm_fx):
+    """
+    Verify a TypeError is thrown, if assigning a non-str value to an optional property.
+    """
+    with pytest.raises(TypeError):
+        test_fm = FactManager(fact_log={"some": "dict"})
+
+
+def test_module_prop_table(test_fm_fx):
+    """
+    Verify private properties are set when passed.
+    """
+    table_name = "some fact table"
+    log_table_name = "some log table"
+
+    test_fm = FactManager(table_name, log_table_name)
+
+    assert test_fm._FACT_TABLE == table_name
+
+
+def test_module_prop_log(test_fm_fx):
+    """
+    Verify private properties are set when passed.
+    """
+    table_name = "some fact table"
+    log_table_name = "some log table"
+
+    test_fm = FactManager(table_name, log_table_name)
+
+    assert test_fm._FACT_LOG == log_table_name
```

Comment:
If this test fails we have much larger problems than this test failing.
From what i can gather there is no possible branch in which this could fail without a broken python install.

Reply:
The init method under test either raises, which isn't being tested here, or succeeds. 
The actual unit under test here is python builtins, specifically setattr and getattr. Neither of which are overloaded

this test is out of scope.

Reply:
I would argue this method also tests the ordering and usage of `__init__`'s arguments.

Reply:
not that we need to test that, but tests are cheap.
/shrug